src/test/app/PermissionedDomains_test.cpp

```diff
@@ -0,0 +1,479 @@
+    void
+    testDelete()
+    {
+        testcase("Delete");
+        Env env(*this, withFeature_);
+        Account const alice("alice");
+
+        env.fund(XRP(1000), alice);
+        auto const setFee(drops(env.current()->fees().increment));
+        pd::Credentials credentials{{alice, pd::toBlob("first credential")}};
+        env(pd::setTx(alice, credentials));
+        env.close();
+        auto objects = pd::getObjects(alice, env);
+        BEAST_EXPECT(objects.size() == 1);
+        auto const domain = objects.begin()->first;
+
+        // Delete a domain that doesn't belong to the account.
+        Account const bob("bob");
+        env.fund(XRP(1000), bob);
+        env(pd::deleteTx(bob, domain), ter(temINVALID_ACCOUNT_ID));
+
+        // Delete a non-existent domain.
+        env(pd::deleteTx(alice, uint256(75)), ter(tecNO_ENTRY));
+
+        // Delete a zero domain.
+        env(pd::deleteTx(alice, uint256(0)), ter(temMALFORMED));
+
+        // Make sure owner count reflects the existing domain.
+        BEAST_EXPECT(pd::ownerInfo(alice, env)["OwnerCount"].asUInt() == 1);
+        auto const objID = pd::getObjects(alice, env).begin()->first;
+        BEAST_EXPECT(pd::objectExists(objID, env));
+        // Delete domain that belongs to user.
+        env(pd::deleteTx(alice, domain), ter(tesSUCCESS));
+        auto const tx = env.tx()->getJson(JsonOptions::none);
+        BEAST_EXPECT(tx[jss::TransactionType] == "PermissionedDomainDelete");
+        // Make sure the owner count goes back to 0.
+        BEAST_EXPECT(pd::ownerInfo(alice, env)["OwnerCount"].asUInt() == 0);
+        // The object needs to be gone.
+        BEAST_EXPECT(pd::getObjects(alice, env).empty());
+        BEAST_EXPECT(!pd::objectExists(objID, env));
+    }
+
+    void
+    testAccountReserve()
+    {
+        // Verify that the reserve behaves as expected for minting.
+        testcase("Account Reserve");
+
+        using namespace test::jtx;
+
+        Env env(*this, withFeature_);
+        Account const alice("alice");
+
+        // Fund alice enough to exist, but not enough to meet
+        // the reserve.
+        auto const acctReserve = env.current()->fees().accountReserve(0);
+        auto const incReserve = env.current()->fees().increment;
+        env.fund(acctReserve, alice);
+        env.close();
+        BEAST_EXPECT(env.balance(alice) == acctReserve);
+        BEAST_EXPECT(pd::ownerInfo(alice, env)["OwnerCount"].asUInt() == 0);
+
+        // alice does not have enough XRP to cover the reserve.
+        pd::Credentials credentials{{alice, pd::toBlob("first credential")}};
+        env(pd::setTx(alice, credentials), ter(tecINSUFFICIENT_RESERVE));
+        BEAST_EXPECT(pd::ownerInfo(alice, env)["OwnerCount"].asUInt() == 0);
+        BEAST_EXPECT(pd::getObjects(alice, env).size() == 0);
+        env.close();
+
+        // Pay alice almost enough to make the reserve.
+        env(pay(env.master, alice, incReserve + drops(19)));
```

Comment:
Please avoid using fee 10 directly, this will break with any fee other than 10.

In this case, this could be written as  
```
auto const baseFee = env.current()->fees().base.drops();
...
env(pay(env.master, alice, incReserve + 2 * baseFee - 1));
```